src/ripple/protocol/Serializer.h

```diff
@@ -280,6 +282,20 @@ class Serializer
     int addEncoded (int length);
 };
 
+template<class FwdIt>
+int Serializer::addVL(FwdIt begin, FwdIt end, int len)
+{
+    int ret = addEncoded(len);
+    if (len)
+        for (; begin != end; ++begin)
```

Comment:
I don't quite like this. Knowing the code base, I known that this is meant to serialize arrays of `base_uint` in one call, but something smells here... 

If nothing else, it doesn't really take a `FwdIt` - only ones that point to something that has `data()` and `size()` methods.